QuizApp/App/Screens/Login/LoginViewModel.swift

```diff
@@ -0,0 +1,38 @@
+import Foundation
+import Combine
+
+final class LoginViewModel {
+
+    @Published var errorLoggingIn = ""
+    private let loginUseCase: LoginUseCaseProtocol
+
+    // MARK: - Init
+
+    init(loginUseCase: LoginUseCaseProtocol) {
+        self.loginUseCase = loginUseCase
+    }
+
+}
+
+extension LoginViewModel {
+
+    func loginUser(password: String, username: String) async {
+        do {
+            let token = try await loginUseCase.loginUser(password, username: username).accessToken
+            /// Save the token to Keychain
+            let keychainService = SecureStorage()
+            keychainService.save(token)
+        } catch {
+            handleError(error)
+        }
+    }
+
+}
+
+private extension LoginViewModel {
```

Comment:
Do you need extension?

Reply:
Just a style I prefer. Even tho it is a simple app, but I prefer to group private methods in a private extension. Though I see value in identifying a method as private at a first glance if it has a private keyword.